### PR TITLE
Test traffic Merge Into Dev: Request for Feedback

### PR DESCRIPTION
**Pull request by sgoggins**
There is one section that looked to me like it might be using an older strategy for adding the traffic gathering stats. Its this one: 

```
### Section from traffic metric merge that may need to be changed

    with DatabaseSession(logger) as session:
        query = session.query(Repo)
        repos = execute_session_query(query, 'all')
        #Just use list comprehension for simple group
        repo_info_tasks = [collect_repo_info.si(repo.repo_git) for repo in repos]

        for repo in repos:
            first_tasks_repo = group(collect_issues.si(repo.repo_git),collect_pull_requests.si(repo.repo_git),collect_github_repo_clones_data.si(repo.repo_git))
            second_tasks_repo = group(collect_events.si(repo.repo_git),
                collect_github_messages.si(repo.repo_git),process_pull_request_files.si(repo.repo_git), process_pull_request_commits.si(repo.repo_git))

            repo_chain = chain(first_tasks_repo,second_tasks_repo)
            issue_dependent_tasks.append(repo_chain)

        repo_task_group = group(
            *repo_info_tasks,
            chain(group(*issue_dependent_tasks),process_contributors.si()),
            generate_facade_chain(logger),
            collect_releases.si()
        )
    
    chain(repo_task_group, refresh_materialized_views.si()).apply_async()

#### End of section from traffic metric merge that may need to be changed
```

@IsaacMilarky  / @ABrain7710  : Let me know if my hunch about that is right. I also believe there's a database change that will be substantially out of order and needs a new name.

### CODE REVIEW

**Comment by sgoggins** [comment]
@ABrain7710 / @IsaacMilarky : Don't bother reviewing this yet ... I can see a bunch of things that require fixing. I will tidy it up first.

File: augur/application/db/data_parse.py
Comment (sgoggins) [inline augur/application/db/data_parse.py:469]: @IsaacMilarky / @ABrain7710 : What needs fixing here?

File: augur/application/db/models/__init__.py
Comment (sgoggins) [inline augur/application/db/models/__init__.py:66]: @IsaacMilarky / @ABrain7710 : is this ok?

File: augur/application/db/models/augur_data.py
Comment (sgoggins) [inline augur/application/db/models/augur_data.py:3363]: @ABrain7710  / @IsaacMilarky : Is this the right way?

File: augur/application/schema/augur_full.sql
Comment (sgoggins) [inline augur/application/schema/augur_full.sql:2779]: @ABrain7710 / @IsaacMilarky : Is this the right way to do this?

File: augur/tasks/init/celery_app.py
Comment (sgoggins) [inline augur/tasks/init/celery_app.py:35]: @IsaacMilarky / @ABrain7710 : This looks right. Just checking.

File: frontend/frontend.config.json
Comment (sgoggins) [inline frontend/frontend.config.json:1]: I think this is fine because to run our frontend, we do still need this file.

**Review by sgoggins — COMMENTED** [review]
@ABrain7710 / @IsaacMilarky : I left several comments. On the frontend related files, I am simply going to run them and see what happens.

File: augur/application/db/models/augur_data.py
Comment (IsaacMilarky) [reply augur/application/db/models/augur_data.py:3363]: There should be a unique constraint on the repo_id if you plan on using postgres 'on conflict' inserts.

File: augur/application/db/data_parse.py
Comment (IsaacMilarky) [reply augur/application/db/data_parse.py:469]: This function LGTM unless it's throwing errors.

File: augur/application/schema/alembic/versions/12_traffic_additions.py
Comment (IsaacMilarky) [inline augur/application/schema/alembic/versions/12_traffic_additions.py:44]: If these changes to the releases table and the user_repos table are intentional then they should probably go in a separate revision. Alembic also sometimes adds spurious edits to the schema depending on the state of the database. Otherwise looks good

File: augur/application/schema/augur_full.sql
Comment (IsaacMilarky) [reply augur/application/schema/augur_full.sql:2779]: The proper way to do this is with alembic which you did already. I would not do it this way.

File: augur/tasks/init/celery_app.py
Comment (IsaacMilarky) [reply augur/tasks/init/celery_app.py:35]: Yep this is correct.

File: augur/tasks/start_tasks.py
Comment (IsaacMilarky) [inline augur/tasks/start_tasks.py:213]: This is old and this new section needs to be removed.

File: augur/tasks/start_tasks.py
Comment (IsaacMilarky) [inline augur/tasks/start_tasks.py:222]: collect_github_repo_clones_data should be moved into the repo_task_group group.

File: frontend/src/views/RepoGroups.vue
Comment (IsaacMilarky) [inline frontend/src/views/RepoGroups.vue:48]: Are we using vue still? This looks pretty old.

**Review by IsaacMilarky — CHANGES_REQUESTED** [review]
Some changes should be made. The section from the merge needs to be removed as it is old, the new table is missing a unique constraint for the on conflict logic, The alembic migration looks like it does a bit more than it should, and the table creation should not be done through augur_full.sql. Otherwise looks good for the most part

File: augur/application/schema/alembic/versions/12_traffic_additions.py
Comment (sgoggins) [inline augur/application/schema/alembic/versions/12_traffic_additions.py:1]: @ABrain7710 / @IsaacMilarky : They are using user_repos in the augur_operations schema as the foreign key. It would be more consistent with our current approach to make the repo table the foreign key.

**Review by sgoggins — COMMENTED** [review]
I think changes are required on the foreign key before this is merged.

File: augur/tasks/start_tasks.py
Comment (sgoggins) [reply augur/tasks/start_tasks.py:77]: @IsaacMilarky : I merged from `dev` into this branch and I think that would fix this .

File: augur/application/db/models/augur_data.py
Comment (sgoggins) [reply augur/application/db/models/augur_data.py:3353]: I did this, and then I realized this is like "releases", or repo_info... we want to hold the historical record for the repos. There should not be any conflicts since the primary key is an autoincrement @IsaacMilarky

**Review by sgoggins — COMMENTED** [review]
I think this is pretty close.

**Comment by sgoggins** [comment]
@IsaacMilarky : I think I addressed this point by merging dev into this branch: https://github.com/chaoss/augur/pull/2220#discussion_r1131361291

**Comment by sgoggins** [comment]
@IsaacMilarky : I might have missed something. Getting this error at runtime... looking into it:
```
[2023-03-19 14:54:45,961: ERROR/ForkPoolWorker-1] Task augur.tasks.start_tasks.augur_collection_monitor[4a790a8d-e655-42d6-b599-337d5f4abc36] raised unexpected: NameError("name 'generate_facade_chain' is not defined")
Traceback (most recent call last):
  File "/home/sean/github/virtualenv/berkeley/lib/python3.8/site-packages/celery/app/trace.py", line 451, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/sean/github/virtualenv/berkeley/lib/python3.8/site-packages/celery/app/trace.py", line 734, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/sean/github/berkeley/augur/tasks/start_tasks.py", line 364, in augur_collection_monitor
    start_primary_collection(session, max_repo=50, days=30)
  File "/home/sean/github/berkeley/augur/tasks/start_tasks.py", line 231, in start_primary_collection
    primary_augur_collection.start_data_collection()
  File "/home/sean/github/berkeley/augur/tasks/util/collection_util.py", line 372, in start_data_collection
    for repo_git, task_id in self.send_messages():
  File "/home/sean/github/berkeley/augur/tasks/util/collection_util.py", line 387, in send_messages
    augur_collection_sequence.append(job(repo_git))
  File "/home/sean/github/berkeley/augur/tasks/start_tasks.py", line 96, in primary_repo_collect_phase
    generate_facade_chain(logger),
NameError: name 'generate_facade_chain' is not defined
```

**Pull request merged**
